src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -32,6 +38,17 @@ private CSharpSyntaxNode ParseTypeOrPatternForIsOperator()
                     break;
             }
 
+            if (tk == SyntaxKind.IdentifierToken && this.CurrentToken.Text == "_")
+            {
+                // In a pattern, we reserve `_` as a wildcard. It cannot be used (with that spelling) as the
+                // type of a declaration or recursive pattern, nor as a type in an in-type expression starting
+                // in C# 7. The binder will give a diagnostic if
+                // there is a usable symbol in scope by that name. You can always escape it, using `@_`.
+                // TODO(patterns2): Should we use the "contextual keyword" infrastructure for this?
```

Comment:
>TODO [](start = 19, length = 4)

This should be a PROTOTYPE comment.

Reply:
This has already been recorded in [https://github.com/dotnet/roslyn/issues/23222](https://github.com/dotnet/roslyn/issues/23222).

---
In reply to: [156497572](https://github.com/dotnet/roslyn/pull/23209#discussion_r156497572) [](ancestors = 156497572)